yt/visualization/plot_modifications.py

```diff
@@ -721,6 +738,13 @@ def __init__(
             plot_args.update(kwargs)
 
         self.plot_args = plot_args
+        if upper_threshold is not None or lower_threshold is not None:
+            threshold_field = threshold_field or self.field
+        elif upper_threshold is None and lower_threshold is None:
+            threshold_field = None  # Need to specify at least one value
```

Comment:
for what it's worth, I find this version much more readable :)

Reply:
Sadly it is also incorrect!  :(